### ISSUE TITLE
Descriptive names for subscriptions

### ISSUE THREAD

**Issue by mmclane** [akuity/kargo#2839]
# Checklist

* [x] I've searched the issue queue to verify this is not a duplicate feature request.
* [x] I've pasted the output of `kargo version`, if applicable.
* [x] I've pasted logs, if applicable.

# Proposed Feature

I would like to be able to give a name to subscriptions in a warehouse.
Today it look like this:
![image](https://github.com/user-attachments/assets/762a25fa-18f9-4097-a645-d54301658c3a)

# Motivation

This would allow them to be more descriptive and make more sense for none-technical folks.

# Suggested Implementation

![image](https://github.com/user-attachments/assets/5663cfd8-8af2-45f0-b17e-0fe8d83dd1af)

**Comment by krancour** [comment]
I agree. fwiw, it's the way it is because every Warehouse and Stage is a Kubernetes resource, which all naturally have names. Subscriptions are just details of a Warehouse, so they do not. Adding an _optional_ name field seems like it would be a nice convenience. Thank you for this suggestion!

**Comment by mmclane** [comment]
I would like to add to this a little bit.  
Today there is only an icon that shows which subscription an artifact comes from in the freight timeline.  If there are two artifacts of the same type but from different subscriptions, that could be confusing and difficult to know what is changing.  I do know that tool tips can be used for this but it definitely could be more clear. If you do have a descriptive name for subscriptions I would like to see that name repeated in the freight timeline.  

Alternately you could use a colored boarder or something to show which subscription each artifact came from.

![image](https://github.com/user-attachments/assets/960a0803-f770-47eb-9ec4-0316c19ccf44)

or something like this could even work.

![image](https://github.com/user-attachments/assets/fe176d32-5a2d-4e2b-bd82-1d733a4c8e12)

Or both?

**Comment by krancour** [comment]
No more colors...

We use colors already to indicate the flow of Freight through a pipeline (the Warehouse border and the connections between nodes) and we also use colors for each Stage so you can easily see what's running where...

And the palette of colors that is aesthetically pleasing AND easily distinguished from one another is limited to such an extent that colors already get reused more than we would like.

If we add colors to more things, it's going to become increasingly _harder_ to see what's actually going on.

**Comment by mmclane** [comment]
I hear that.. but something needs to tie those together.

**Comment by mmclane** [comment]
Take this example.  What freight goes with which warehouse.  Which subscription matches with each artifact?

![image](https://github.com/user-attachments/assets/8d25603d-7134-4aba-abf9-d57eb2e1bfe0)

**Comment by mmclane** [comment]
This is even worse because it doesn't even show all of the artifacts.

![image](https://github.com/user-attachments/assets/70e517d2-ba44-4db1-9dbf-91127ab8433a)

**Comment by krancour** [comment]
There's already a separate issue to color Freight borders with the Warehouse color. That's fine because it correlates one thing to something else that is already color-coded.

**Comment by Marvin9** [comment]
I have improvements that could resolve the subscription visibility issue. I agree with @krancour that colours are not solution rather it could lead to confusion.

1. Increase the Warehouse dropdown reachability. Right now it is far right side on screen but I think if we put it in left-most, the readability will increase.
2. Highlight the selected warehouse from dropdown (if selected individual). Right now if we select warehouse from dropdown, there is filter icon in the pipeline but I believe its not enough and needs more "highlight" design.
3. Re: When there are more freights, the box adds scrollbar that leaves last freight invisible at quick sight
  - Short term solution: Let the height flow naturally to fit all the freight in box
  - Long term solution: I would split the freight timeline into rows, each row will have distinct freights (1st row git, 2nd row images...). The column logically aligns the freight same as it is right now but splitting into row would open doors for flexibility for different choices of view like show only git or keep the container images in first row)

**Comment by krancour** [comment]
@Marvin9 mostly agree with everything, but have some skepticism about:

> Long term solution: I would split the freight timeline into rows, each row will have distinct freights (1st row git, 2nd row images...). The column logically aligns the freight same as it is right now but splitting into row would open doors for flexibility for different choices of view like show only git or keep the container images in first row)

We've seen cases in the wild where Warehouses subscribe to _lots_ of different repos. I don't necessarily think that's a good idea because it's naturally quite hard for humans to comprehend the sum total of what's in a piece of Freight when there are dozens of artifacts within... but this _is_ a thing that some users are doing. With this in mind, a row per subscription in the Freight Timeline may eat up all vertical space in the viewport and then some.

**Comment by Marvin9** [comment]
Oh wait, I meant the same freight timeline view as it is now but just more connected dimensions (as of now its just freight box with freights), once we split into rows there will be common stretched box (say for same git hash for 3-4 consecutive freights). I will put some drawing

**Comment by krancour** [comment]
_Tentatively_ in v1.11.0 since @mail2sudheerobbu-oss has opened a PR.

**Comment by krancour** [comment]
Additional context in this dupe: https://github.com/akuity/kargo/issues/5729

**Comment by krancour** [comment]
#6207 took a big bite out of this issue and the things it _didn't_ address now have more narrowly-scoped follow-up issues #6724 and #6729, so I think it reasonable to close this issue now in favor of those.
